### ISSUE TITLE
Provision a MacOS box for running unit and e2e tests

### ISSUE THREAD

**Issue by khaledsulayman** [instructlab/instructlab#1112]
**Is your feature request related to a problem? Please describe.**
IIUC, the current actions workflow only runs on Ubuntu. Certain jobs such as the [e2e job](https://github.com/instructlab/instructlab/actions/workflows/e2e.yml) may behave differently between MacOS and Linux and this could lead to us missing breaking changes. 

**Describe the solution you'd like**
Provision a box running MacOS that we can run unit and e2e tests on. Hopefully it wouldn't be too much more costly since it doesn't need to run all actions. Certain actions like lint and format as they shouldn't differ between systems.

**Comment by tiran** [comment]
I think that the default macos runner should be good enough to test training on MPS / METAL. Let's give it a try.

**Comment by tiran** [comment]
Looks like GHA does not support MPS. https://github.com/actions/runner-images/issues/9254 suggests it's a limitation in Apple's virtualization:

> it is not a GitHub restriction, Apple does not support nested virt on M1 CPU gen entirely

```raw
ggml_metal_graph_compute_block_invoke: error: unsupported op 'RMS_NORM'
GGML_ASSERT: /private/var/folders/3m/p59k4qdj0f17st0gn2cmj3640000gn/T/pip-install-abmja2gi/llama-cpp-python_94634b10a2fc48d9acb183bde5c3d2c4/vendor/llama.cpp/ggml-metal.m:828: !"unsupported op"
```

**Comment by nathan-weinberg** [comment]
I think this is still worth having, though it may be a while longer before we can implement

**Comment by bjhargrave** [comment]
We use EC2 for e2e tests. Can we set up one for a `mac2-m2.metal` instance?

**Comment by jaideepr97** [comment]
closing in favor of #1638
